### PR TITLE
Implement conversion of stablehlo.maximum OP to TTIR dialect

### LINKED ISSUE #508: [MNIST] Implement conversion of stablehlo.maximum OP to TTIR dialect
`stablehlo.maximum` OP specification can be found [here](https://openxla.org/stablehlo/spec#maximum).

Seems like we need to add support in TTIR for MaximumOP. It can be lowered into [`ttnn.maximum`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/maximum.html).

### LINKED ISSUE #546: [Ops] Implement Maximum op

### PR DESCRIPTION
**Pull request by mrakitaTT**
Added MaximumOP to TTIR and TTNN dialects and implemented conversion from stablehlo.maximum OP to ttir.maximum OP.

Added necessary tests.

Fixes #508
Fixes #546

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]
fyi @jnie-TT for runtime changes

File: runtime/lib/ttnn/program.cpp
Comment (jnie-TT) [inline runtime/lib/ttnn/program.cpp:200]: @mrakitaTT thanks! This definitely removes alot of boilerplate code, but my worry is that we may want different configs/parameters for different ops. For example I see that binary ops in ttnn accept scalar values; there's also implicit broadcasting for associative binary ops, which may or may not affect the constraints on the memory configs of the input tensors. Therefore I'm leaning towards leaving the structure unchanged for now, and once we have well-documented op constraints then we can merge/generalize the code. What do you think?

Comment (sdjordjevicTT) [reply runtime/lib/ttnn/program.cpp:200]: I like this refactor. @jnie-TT, we can easily pass additional configs through these new functions if needed, right?

File: test/ttmlir/Conversion/StableHLOToTTIR/mnist_inference.mlir
Comment (sdjordjevicTT) [inline test/ttmlir/Conversion/StableHLOToTTIR/mnist_inference.mlir:1]: Do we have all the necessary support and required conversions in place to run this?

File: runtime/lib/ttnn/program.cpp
Comment (mrakitaTT) [reply runtime/lib/ttnn/program.cpp:200]: @jnie-TT Thanks for pointing that out, I understand the concern, but IMO it is always better to start with generalized compact code and then only if it turns out that we need bunch of specific customizations to go from there and expand the code. As @sdjordjevicTT said it might be fairly easy to pass additional configs to these functions, but even if it ends up being not so easy for certain OPs then we can have custom functions for those OPs only. Budabackend stack was littered with duplicated code and I would really like us to change that mindset for the future stacks :)

File: test/ttmlir/Conversion/StableHLOToTTIR/mnist_inference.mlir
Comment (mrakitaTT) [reply test/ttmlir/Conversion/StableHLOToTTIR/mnist_inference.mlir:1]: Not yet, we are currently working toward it so I wanted to commit it for the reference. Do I need to put the `// UNSUPPORTED: true` in the test? I've noticed that in a few tests, is that used by the lit to skip the test?

Comment (sdjordjevicTT) [reply test/ttmlir/Conversion/StableHLOToTTIR/mnist_inference.mlir:1]: Yeah, that's why I asked in the first place, please add:
`// UNSUPPORTED: true`

This will mark a test not to run, once we add a support for it, we can remove the tag.

**Review by sdjordjevicTT — APPROVED** [review]

File: lib/Conversion/StableHLOToTTIR/StableHLOToTTIR.cpp
Comment (mrakitaTT) [reply lib/Conversion/StableHLOToTTIR/StableHLOToTTIR.cpp:87]: Yeah, it was weird to me too. It is interesting how their reduce works, they have a generic reduce OP which they combine with binary OPs. For example to do what `ttnn.max` does, they use some initial value and then apply reduce with maximum on tensor by doing maximum(initValue, tensor[0]), then maximum(result, tensor[1]), etc :)
```
%54 = stablehlo.reduce(%52 init: %53) applies stablehlo.maximum across dimensions = [1] : (tensor<128x10xf32>, tensor<f32>) -> tensor<128xf32>
```

Comment (svuckovicTT) [reply lib/Conversion/StableHLOToTTIR/StableHLOToTTIR.cpp:87]: Looks peculiar :D Do you know what do they do if the tensor is > 2d?

Comment (mrakitaTT) [reply lib/Conversion/StableHLOToTTIR/StableHLOToTTIR.cpp:87]: Yeah, that's where `across dimensions = [1]` comes from, tensor can be of any rank but you specify the dimension across which you want to do reduce.

File: runtime/lib/ttnn/program.cpp
Comment (mrakitaTT) [reply runtime/lib/ttnn/program.cpp:200]: Hi @jnie-TT do you still have concerns about this? I would like to proceed with merge, just waiting on your sign-off.

**Review by jnie-TT — APPROVED** [review]

**Pull request merged**
